### ISSUE TITLE
Use Self in constructors [ECR-1377]

### ISSUE THREAD

**Issue by vitvakatu** [exonum/exonum#672]
At the moment we have a lot of 
```rust
fn new() -> MyType { MyType {...} }
```
constructors.

They should be replaced with 
```rust
fn new() -> Self { Self {...} }
```

**Comment by aleksuss** [comment]
I see one significant profit from renaming into the `Self`. If we will decide to rename struct we'll need to do less refactoring. It's a good reason i suppose.

**Comment by DarkEld3r** [comment]
Closed with #751.
